File: ui/src/api/infra/migration.ts
import {http} from '@/utils/http'
import { BASE_URL, type ApiResponse } from '@/api/utils'

type MigrationResult = {
  status: string
  message: string
  total: number
  success: number
  failed: number
  errors?: string[]
}

export const importMarkdown = (files: File[]) => {
  const formData = new FormData()
  files.forEach(file => {
    formData.append('files', file)
  })
  return http.request<ApiResponse<MigrationResult>>('post', `${BASE_URL}/v1/migration/md`, {
    data: formData,
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}

File: pkg/domain/model/migration.go
package model

type MigrationResult struct {
	Status  string   `json:"status"`
	Message string   `json:"message"`
	Total   int      `json:"total"`
	Success int      `json:"success"`
	Failed  int      `json:"failed"`
	Errors  []string `json:"errors,omitempty"`
}
